server/src/main/java/org/apache/iotdb/db/newsync/pipedata/queue/PipeDataQueue.java

```diff
@@ -0,0 +1,17 @@
+package org.apache.iotdb.db.newsync.pipedata.queue;
+
+import org.apache.iotdb.db.newsync.pipedata.PipeData;
+
+import java.util.List;
+
+public interface PipeDataQueue {
+    boolean offer(PipeData data);
+
+    List<PipeData> pull(long serialNumber);
+
+    PipeData take() throws InterruptedException;
+
+    void commit();
+
+    void clear();
```

Comment:
should add function `close` to close stream but not delete log directory.

Reply:
It seems not necessary now.